### ISSUE TITLE
fix(external-link): URLs containing parentheses (e.g. DOI) are truncated during extraction

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#273]
## Problem

URLs containing parentheses — such as DOI URLs — are truncated during link extraction because the regex treats `)` unconditionally as a URL terminator.

**Example:**

```
https://doi.org/10.1016/0026-0495(67)90027-3
```

is extracted as two broken fragments:

```
https://doi.org/10.1016/0026-0495(67
https://doi.org/10.1016/0026-0495
```

Reported in: #147 (comment)

## Root cause

`internal/rule/external_link.go` L13–15:

```go
inlineLinkPattern = regexp.MustCompile(`\[[^\]]*\]\((https?://[^\s)]+)\)`)
imageLinkPattern  = regexp.MustCompile(`!\[[^\]]*\]\((https?://[^\s)]+)\)`)
bareURLPattern    = regexp.MustCompile(`(?m)^.*?(https?://[^\s<>"'()]+).*?$`)
```

`[^\s)]+` excludes `)` from URLs entirely. The CommonMark spec allows parentheses in URLs as long as they are balanced — a `)` that has a matching `(` within the URL is part of the URL, not a terminator.

## Proposed fix

Allow `(...)` groups within the captured URL portion:

```go
// ( ) pairs inside the URL are allowed; unmatched ) still terminates
inlineLinkPattern = regexp.MustCompile(`\[[^\]]*\]\((https?://(?:[^\s()\[\]]+|\([^\s()]*\))*)\)`)
imageLinkPattern  = regexp.MustCompile(`!\[[^\]]*\]\((https?://(?:[^\s()\[\]]+|\([^\s()]*\))*)\)`)
bareURLPattern    = regexp.MustCompile(`(?m)^.*?(https?://(?:[^\s<>"'()]+|\([^\s<>"'()]*\))+).*?$`)
```

The sub-pattern `(?:[^\s()\[\]]+|\([^\s()]*\))*` matches either regular URL characters or a balanced `(...)` group, supporting one level of nesting — sufficient for DOI and most real-world URLs with parentheses.

## Acceptance criteria

- [ ] `https://doi.org/10.1016/0026-0495(67)90027-3` is extracted as a single URL
- [ ] Standard Markdown links `[text](https://example.com)` still extract correctly
- [ ] Image links `![alt](https://example.com/img.png)` still extract correctly
- [ ] Bare URLs without parentheses are unaffected
- [ ] A URL ending with an unmatched `)` is still terminated at that `)`
- [ ] Unit tests added for DOI-style URLs and other parenthesized URL patterns
